kategory-docs/docs/docs/datatypes/ior/README.md

```diff
@@ -6,4 +6,125 @@ permalink: /docs/datatypes/ior/
 
 ## Ior 
 
-TODO
\ No newline at end of file
+`Ior` represents an inclusive-or relationship between two data types.
+This makes it very similar to the [`Either`](/docs/datatypes/either) data type, which represents an "exclusive-or" relationship.
+What this means, is that an `Ior<A, B>` (also written as `A Ior B`) can contain either an `A`, a `B`, or both an `A` and `B`.
+Another similarity to `Either` is that `Ior` is right-biased,
+which means that the `map` and `flatMap` functions will work on the right side of the `Ior`, in our case the `B` value.
+You can see this in the function signature of `map`:
+
+```kotlin
+fun <D> map(f: (B) -> D): Ior<A, D>
+```
+
+We can create `Ior` values using `Ior.left`, `Ior.right` and `Ior.both`:
+
+```kotlin:ank
+import kategory.*
+
+val right = Ior.Right(3)
+
+val left = Ior.Left("Error")
+
+val both = Ior.Both("Warning", 3)
+```
+
+Kategory also offers extension functions for `Ior`, the `leftIor`, `rightIor` and `bothIor`:
+
+```kotlin:ank
+import kategory.*
+
+val right = 3.rightIor()
+
+val left = "Error".leftIor()
+
+val both = ("Warning" to 3).bothIor()
+```
+
+
+When we look at the `Monad` or `Applicative` instances of `Ior`, we can see that they actually require a `Semigroup` instance on the left side.
+This is because `Ior` will actually accumulate failures on the left side, very similarly to how the [`Validated`](/docs/datatypes/validated) data type does.
+This means we can accumulate data on the left side while also being able to short-circuit upon the first right-side-only value.
+For example, sometimes, we might want to accumulate warnings together with a valid result and only halt the computation on a "hard error"
+Here's an example of how we are able to do that:
+
+```kotlin:ank
+import kategory.*
+
+data class User(val name: String, val pw: String)
+
+fun validateUsername(username: String) = when {
+    username.isEmpty() -> Nel.of("Can't be empty").leftIor()
+    username.contains(".") -> (Nel.of("Dot in name is deprecated") to username).bothIor()
+    else -> username.rightIor()
+}
+
+
+fun validatePassword(password: String) = when {
+    password.length < 8 -> Nel.of("Password too short").leftIor()
+    password.length < 10 -> (Nel.of("Password should be longer") to password).bothIor()
+    else -> password.rightIor()
+}
+
+fun validateUser(name: String, pass: String) =
+        Ior.monad<Nel<String>>().binding {
+            val username = validateUsername(name).bind()
+            val password = validatePassword(pass).bind()
+            yields(User(username, password))
+        }.ev()
+
+```
+
+Now we're able to validate user data and also accumulate non-fatal warnings:
+
+```kotlin:ank
+
+validateUser("John", "password12")
```

Comment:
for ank to print the result of this expressions they should be added into separate snippets. Ank can only evaluate a snippet as a whole.